### PR TITLE
Fixes #12127 - sets z-index on block list actions element

### LINKED ISSUE #12127: Block List actions overlay is invisible if custom view uses z-index
### Which *exact* Umbraco version are you using? For example: 9.0.1 - don't just write v9

9.3.1

### Bug summary

If you have a block with a custom view that uses z-index in any way, the actions overlay (delete, copy) becomes invisible because it has no z-index specified. adding a simple `z-index: 500` to `.umb-block-list__block--actions` fixes the issue without any side effects that I can detect. Are there any guidelines as to which z-index values are used for which areas in the backoffice and if so, is 500 a good value to use here?

### Specifics

_No response_

### Steps to reproduce

Create a block list editor with a block type that has a custom view.
In that custom view, use CSS to style a block with `z-index: 1` and make sure that it covers the whole editor area, specially the top right.
Notice how the delete/copy buttons are now invisible.

### Expected result / actual result

The buttons should remain visible, no matter what my CSS for the block preview does.

### PR DESCRIPTION
**Pull request by jannikanker**
Should solve [#12127](https://github.com/umbraco/Umbraco-CMS/issues/12127)

### Prerequisites

- [ ] I have added steps to test this contribution in the description below

If there's an existing issue for this PR then this fixes https://github.com/umbraco/Umbraco-CMS/issues/12127

### Description
As described in the issue, this fix helps to ensure that actions for a block list editor remain visible, even if a custom view uses z-index (as long as the custom view doesn't go above z-index: 500, I guess).

### CODE REVIEW

**Comment by mikecp** [comment]
Hi @jannikanker , thanks for this fix !

Before merging, I'd like to check with the front-end team at HQ if 500 is enough and/or if we should better use another value, maybe one of the variables in the .less files.

Let me check with them and come back to you 😊

Cheers!

**Comment by jannikanker** [comment]
@mikecp Excellent idea :-) I took a quick look at what other z-indexes are used, and they are (seemingly) all over the place :-D So I figured 500 was actually a good number, but yeah, maybe the frontend team has a better number.

**Comment by bjarnef** [comment]
Maybe we should add this as a variable here to get overview of the index compared to other z-indexes:
https://github.com/umbraco/Umbraco-CMS/blob/v10/contrib/src/Umbraco.Web.UI.Client/src/less/variables.less#L354-L374

**Comment by mikecp** [comment]
Thanks @bjarnef !
Yes exactly, but I first wanted to check with HQ if it would be appropriate to use one of the existing ones. But if not, we'll definitely need to add one in there.
I've asked the question, now waiting for the answer 🙂

**Comment by mikecp** [comment]
Hey @jannikanker ,

Apologies for the delayed answer. I checked this with HQ and we'd go for the creation of a new z-index variable in the . less file, as suggested by @bjarnef .

Cheers!

**Comment by jannikanker** [comment]
@mikecp @bjarnef Cool! Any thoughts on naming?

**Comment by mikecp** [comment]
Something like @zIndexBlockListActions ?

**Comment by bjarnef** [comment]
`@zIndexBlockActions` as it is used in Block Grid editor as well and on block cards in configuration.

**Comment by jannikanker** [comment]
PR updated :-)

**Review by mikecp — APPROVED** [review]

**Comment by mikecp** [comment]
Thanks @jannikanker , all tested and working fine, so let's merge this 🎉
And thanks @bjarnef for your input 👍

Cheers!

**Pull request merged**
